### PR TITLE
Frontend: React + Bootstrap, Backend database: Postgresql

### PR DESCRIPTION
**Pull request by Dzheims**

### CODE REVIEW

File: Backend/services/student/index.js [hunk 1]
```diff
@@ -1,49 +1,86 @@
...
-    })
-})
+  const id = parseInt(request.params.id);
+  pool.query(queries.getStudentById, [id], (error, result) => {
+    if (error) throw error;
+
+    if (result.rows.length) {
+      response.status(200).json(result.rows[0]);
+    } else {
+      response.send('Student does not exist');
```
Comment (24thsaint) [inline Backend/services/student/index.js:28]: Use `response.status(404).send('Student does not exist')`

This also occurs multiple times throughout your implementation. Observe good http status codes. :))

File: Backend/services/student/index.js [hunk 2]
```diff
@@ -1,49 +1,86 @@
...
  * Returns a single student in the database.
  */
 router.get('/:id', (request, response) => {
-    response.status(500).send({
-        error: 'Pending Implementations'
-    })
-})
+  const id = parseInt(request.params.id);
+  pool.query(queries.getStudentById, [id], (error, result) => {
+    if (error) throw error;
```
Comment (24thsaint) [inline Backend/services/student/index.js:23]: Use `if (error) return response.status(500).send(error.message)`

This also occurs multiple times throughout your implementation. Observe good http status codes and handling of general errors. :))

File: Backend/services/user/index.js [hunk 3]
```diff
@@ -1,44 +1,108 @@
...
+  const { firstName, lastName, username, password } = request.body;
+
+  pool.query(queries.checkUserExists, [username], (error, result) => {
+    if (error) throw error;
+
+    if (result.rows.length) {
+      if (error) throw error;
+      response.send('Username already exists');
+    } else {
+      hashPassword(password, (hashError, hashedPassword) => {
```
Comment (24thsaint) [inline Backend/services/user/index.js:61]: Wonderful.

File: Backend/services/user/index.js [hunk 4]
```diff
@@ -1,44 +1,108 @@
...
-        error: 'Pending Implementations'
-    })
-})
+  const { firstName, lastName, username, password } = request.body;
+
+  pool.query(queries.checkUserExists, [username], (error, result) => {
+    if (error) throw error;
+
+    if (result.rows.length) {
+      if (error) throw error;
```
Comment (24thsaint) [inline Backend/services/user/index.js:58]: Unnecessary repetition of `if (error) throw error;`

File: Frontend/src/views/Components/Sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+  SidebarHeader,
+  SidebarFooter,
+  SidebarContent,
+} from 'react-pro-sidebar';
+import { Link, useNavigate } from 'react-router-dom';
+import { Card, Row, Col } from 'react-bootstrap';
+import { sidebarContext } from './SidebarContext';
+import Cookies from 'js-cookie';
+
+const menuItem = require('./sidebarItems.json');
```
Comment (24thsaint) [inline Frontend/src/views/Components/Sidebar/Sidebar.jsx:18]: Good work.

File: Frontend/src/views/Components/Sidebar/Sidebar.jsx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+      style={{
+        padding: '0px',
+      }}
+    >
+      <div
+        style={{
+          whiteSpace: 'nowrap',
+          display: 'flex',
+          flexDirection: 'column',
+          // alignItems: `${isOpen ? 'flex-end' : 'flex-start'}`,
```
Comment (24thsaint) [inline Frontend/src/views/Components/Sidebar/Sidebar.jsx:57]: Don't leave commented out code. :))

If necessary, leave an explanation why it's left commented.
If unnecessary, do not hesitate to remove.

File: Frontend/src/views/Login.jsx [hunk 7]
```diff
@@ -1,38 +1,102 @@
...
+            'user_name',
+            `${response.data.user.firstname} ${response.data.user.lastname}`
+          );
+        }
+      })
+      .then(() => {
+        navigate('/dashboard');
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (24thsaint) [inline Frontend/src/views/Login.jsx:73]: Handle frontend errors properly (eg. Use toasts to let the user know what happened).

**Review by 24thsaint — APPROVED** [review]
Hi @Dzheims ,

Thank you for attempting the extra credits. 💯 

Your implementation is good. You were able to execute what we need. Good grasp of how to use the language, frameworks, and libraries.

Your error handling habits need a little bit of work and the biggest flaw in the application is the improper use of cookies and handling login sessions. Using the `first name` and `last name` alone as the cookie key is dangerous. 😅 Next time, try generating session hashes or use JWT (jwt.io) for cookie session management.

I hope you'll learn this with us and improve as you move forward on your software engineering journey.

_PS. No need to re-submit corrections, just take note of the comments for your future projects. 🎉_

**Comment by Dzheims** [comment]
Hello sir Rave.Thank you for your corrections and advice, it is very much appreciated. I hope to work and learn a lot more from you moving forward.

**Pull request closed without merging**